lib/internal/streams/from.js

```diff
@@ -37,15 +37,26 @@ function from(Readable, iterable, opts) {
   // Reading boolean to protect against _read
   // being called before last iteration completion.
   let reading = false;
+  let needToClose = false;
   readable._read = function() {
     if (!reading) {
       reading = true;
       next();
     }
   };
+  readable._destroy = function() {
+    if (needToClose) {
```

Comment:
Do you even need `needToClose` why not always call return?

Reply:
> Do you even need `needToClose` why not always call return?

@ronag 
If iterator returns `{ done = true }` then I should not call `.return()`
It can be no-op, but for I do this for consistency with how it handles by `for-of` loop

Reply:
`.return()` should not be called also if iterator throws on `.next()`